assert/assertions.go

```diff
@@ -2045,24 +2045,50 @@ func ErrorAs(t TestingT, err error, target interface{}, msgAndArgs ...interface{
 		return true
 	}
 
-	chain := buildErrorChainString(err)
+	chain := buildErrorChainString(err, true)
 
 	return Fail(t, fmt.Sprintf("Should be in error chain:\n"+
-		"expected: %q\n"+
+		"expected: %T\n"+
 		"in chain: %s", target, chain,
 	), msgAndArgs...)
 }
 
-func buildErrorChainString(err error) string {
+func unwrapAll(err error) (errs []error) {
+	errs = append(errs, err)
```

Comment:
Do you mind explaining this one? I believe this should only be the default cause since you're going to have recursive definitions of error if you explore each value in the slice.

Reply:
This basically "flattens" the provided error (`err`) into a slice so we can access the type and `Error()` string of each constituent error. We start by appending `err` to the slice because it itself has a type and `Error()` string.

`errors.As` failing for a given error means that the target error type is not present in any of these constituent errors. So, we need to exhaustively log all of the types in the error message.

I'm not 100% sure if this answers your question.

Reply:
Aaa, so the usecase I am thinking of is the following:

```golang
errs := unwrap(errors.Join(errors.New("a"), errors.New("b"))
```
This would mean the resulting values would be:

```golang
[]error(error([]{"a", "b"}, "a", "b")
```

Which I don't believe you want to repeat those values

Reply:
In this case, I do want to repeat those values - I want 3 lines in the error message, and 3 types.